lib/core/src/data/function/nop.rs

```diff
@@ -0,0 +1,37 @@
+// ===========
+// === NOP ===
+// ===========
+
+pub trait NOP {
```

Comment:
This was for implementing Default for callbacks. As you noted, we are not using it extensively now, but it would be very useful when defining the builder pattern here. If you've got a callback in a builder pattern, a default value for it which just does not do anything is useful.

Reply:
Right, after reading rest of the code I see the point